client/app/scripts/components/timeline-control.js

```diff
@@ -0,0 +1,251 @@
+
+import {
+  TIMELINE_SLIDER_UPDATE_INTERVAL,
+  TIMELINE_DEBOUNCE_INTERVAL,
+} from '../constants/timer';
+
+
+const sliderRanges = {
+  last15Minutes: {
+    label: 'Last 15 minutes',
+    getStart: () => moment().utc().subtract(15, 'minutes'),
+  },
+  last1Hour: {
+    label: 'Last 1 hour',
+    getStart: () => moment().utc().subtract(1, 'hour'),
+  },
+  last6Hours: {
+    label: 'Last 6 hours',
+    getStart: () => moment().utc().subtract(6, 'hours'),
+  },
+  last24Hours: {
+    label: 'Last 24 hours',
+    getStart: () => moment().utc().subtract(24, 'hours'),
+  },
+  last7Days: {
+    label: 'Last 7 days',
+    getStart: () => moment().utc().subtract(7, 'days'),
+  },
+  last30Days: {
+    label: 'Last 30 days',
+    getStart: () => moment().utc().subtract(30, 'days'),
+  },
+  last90Days: {
+    label: 'Last 90 days',
+    getStart: () => moment().utc().subtract(90, 'days'),
+  },
+  last1Year: {
+    label: 'Last 1 year',
+    getStart: () => moment().subtract(1, 'year'),
+  },
+  todaySoFar: {
+    label: 'Today so far',
+    getStart: () => moment().utc().startOf('day'),
+  },
+  thisWeekSoFar: {
+    label: 'This week so far',
+    getStart: () => moment().utc().startOf('week'),
+  },
+  thisMonthSoFar: {
+    label: 'This month so far',
+    getStart: () => moment().utc().startOf('month'),
+  },
+  thisYearSoFar: {
+    label: 'This year so far',
+    getStart: () => moment().utc().startOf('year'),
+  },
+};
+
+class TimelineControl extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      showSliderPanel: false,
+      millisecondsInPast: 0,
+      rangeOptionSelected: sliderRanges.last1Hour,
+    };
+
+    this.renderRangeOption = this.renderRangeOption.bind(this);
+    this.handleTimestampClick = this.handleTimestampClick.bind(this);
+    this.handleJumpToNowClick = this.handleJumpToNowClick.bind(this);
+    this.handleSliderChange = this.handleSliderChange.bind(this);
+    this.debouncedUpdateTimestamp = debounce(
+      this.updateTimestamp.bind(this), TIMELINE_DEBOUNCE_INTERVAL);
+  }
+
+  componentWillMount() {
+    // Force periodic re-renders to update the slider position as time goes by.
+    this.timer = setInterval(() => { this.forceUpdate(); }, TIMELINE_SLIDER_UPDATE_INTERVAL);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer);
+    this.updateTimestamp();
+  }
+
+  handleSliderChange(sliderValue) {
+    let millisecondsInPast = this.getRangeMilliseconds() - sliderValue;
+
+    // If the slider value is less than 1s away from the right-end (current time),
+    // assume we meant the current time - this is important for the '... so far'
+    // ranges where the range of values changes over time.
+    if (millisecondsInPast < 1000) {
+      millisecondsInPast = 0;
+    }
+
+    this.setState({ millisecondsInPast });
+    this.debouncedUpdateTimestamp(millisecondsInPast);
+    this.props.startWebsocketTransition();
```

Comment:
No, the only thing it does is to start the loader. But that's a good but, it's not named that well so I'll rename it to `startWebsocketTransitionLoader`.